flake-modules/updates/rust-analyzer.nix

```diff
@@ -0,0 +1,201 @@
+
+  # Extract only the generated properties, removing vscode specific options
+  rustAnalyzerProperties = lib.lists.sublist (generatedStart + 1) (
+    generatedEnd - generatedStart - 1
+  ) options;
+
+  mkRustAnalyzerOptionType =
+    nullable: property_name:
+    {
+      type,
+      enum ? null,
+      minimum ? null,
+      maximum ? null,
+      items ? null,
+      anyOf ? null,
+      # Not used in the function, but anyOf values contain it
+      enumDescriptions ? null,
+    }@property:
+    if enum != null then
+      {
+        kind = "enum";
+        values = enum;
+      }
+    else if anyOf != null then
+      let
+        possibleTypes = lib.filter (sub: !(sub.type == "null" && nullable)) anyOf;
+      in
+      {
+        kind = "oneOf";
+        subTypes = builtins.map (
+          t: mkRustAnalyzerOptionType nullable "${property_name}-sub" t
+        ) possibleTypes;
+      }
+    else if lib.isList type then
+      (
+        if lib.head type == "null" then
+          assert lib.length type == 2;
+          let
+            innerType = property // {
+              type = lib.elemAt type 1;
+            };
+            inner = mkRustAnalyzerOptionType nullable "${property_name}-inner" innerType;
+          in
+          # If this assert is false we need to emit a nullOr type
+          assert nullable;
+          inner
+        else
+          let
+            innerTypes = builtins.map (
+              t: mkRustAnalyzerOptionType nullable "${property_name}-inner" (property // { type = t; })
+            ) type;
+          in
+          {
+            kind = "oneOf";
+            subTypes = innerTypes;
+          }
+      )
+    else if type == "array" then
+      {
+        kind = "list";
+        item = mkRustAnalyzerOptionType false "${property_name}-item" items;
+      }
+    else if type == "number" || type == "integer" then
+      {
+        kind = type;
+        inherit minimum maximum;
+      }
+    else if
+      lib.elem type [
+        "object"
+        "string"
+        "boolean"
+      ]
+    then
+      { kind = type; }
+    else
+      throw "Unhandled value in ${property_name}: ${lib.generators.toPretty { } property}";
+
+  mkRustAnalyzerOption =
+    property_name:
+    {
+      # List all possible values so that we are sure no new values are introduced
+      default,
+      markdownDescription,
+      enum ? null,
+      enumDescriptions ? null,
+      anyOf ? null,
+      minimum ? null,
+      maximum ? null,
+      items ? null,
+      # TODO: add this in the documentation ?
+      uniqueItems ? null,
+      type ? null,
+    }:
+    let
+      enumDesc =
+        values: descriptions:
+        let
+          valueDesc = builtins.map ({ fst, snd }: ''- ${fst}: ${snd}'') (
+            lib.lists.zipLists values descriptions
+          );
+        in
+        ''
+          ${markdownDescription}
+
+          Values:
+          ${builtins.concatStringsSep "\n" valueDesc}
+
+          default:
```

Comment:
`pluginDefault` should be separate from description.

The shim should pass `pluginDefault` through `helpers.pluginDefaultText` to be consistent with our defaultNullOpts.

```nix
mkOption {
  defaultText = helpers.pluginDefaultText {
    inherit pluginDefault;
    default = null;
  };
}
```